### PR TITLE
[build] Actually remove ENABLE_WALLET

### PR DESCRIPTION
**Pull request by jnewbery**
Adds a couple of redefinitions to dummywallet.cpp.

### CODE REVIEW

**Comment by jnewbery** [comment]
Builds on #14204. Review that first.

**Comment by laanwj** [comment]
utACK abdba5f59e348c7093bee2639ef6c2398e8d2f6c

File: src/interfaces/node.cpp [hunk 1]
```diff
@@ -221,15 +219,15 @@ class NodeImpl : public Node
     }
     std::vector<std::unique_ptr<Wallet>> getWallets() override
     {
-#ifdef ENABLE_WALLET
+        if (!g_wallet_init_interface.HasWalletSupport()) {
+            throw std::logic_error("Node::getWallets() called in non-wallet build.");
```
Comment (ryanofsky) [inline src/interfaces/node.cpp]: Can you drop g_wallet_init_interface references and throws in this file? They seem redundant if MakeWallet and GetWallets functions can throw now.

Comment (jnewbery) [reply src/interfaces/node.cpp]: Done!

**Comment by jnewbery** [comment]
#14204 is merged. I've rebased on master and addresed @ryanofsky's feedback: https://github.com/bitcoin/bitcoin/pull/14208#discussion_r217068662

**Comment by MarcoFalke** [comment]
utACK 3efef4b225308954b23a30c74d7781ba87a51b76

**Review by scravy — APPROVED** [review]
utACK 3efef4b225308954b23a30c74d7781ba87a51b76

**Review by ryanofsky — COMMENTED** [review]
utACK 3efef4b225308954b23a30c74d7781ba87a51b76

File: src/interfaces/node.cpp [hunk 2]
```diff
@@ -32,18 +32,15 @@
...
-#else
-#define CHECK_WALLET(x) throw std::logic_error("Wallet function called in non-wallet build.")
-#endif
 
 #include <atomic>
 #include <boost/thread/thread.hpp>
 #include <univalue.h>
 
+class CWallet;
+class Wallet;
```
Comment (practicalswift) [inline src/interfaces/node.cpp]:
```
2018-09-12 23:23:50 clang-tidy(pr=14208): interfaces/node.cpp:41:7: warning: declaration 'Wallet' is never referenced, but a declaration with the same name found in another namespace 'interfaces' [bugprone-forward-declaration-namespace]
2018-09-12 23:23:50 clang-tidy(pr=14208): interfaces/node.cpp:41:7: warning: no definition found for 'Wallet', but a definition with the same name 'Wallet' found in another namespace 'interfaces' [bugprone-forward-declaration-namespace]
```

Comment (promag) [reply src/interfaces/node.cpp]: Yap, should be fixed.

Comment (jnewbery) [reply src/interfaces/node.cpp]: fixed

**Comment by jnewbery** [comment]
I've addressed the comment in https://github.com/bitcoin/bitcoin/pull/14208#discussion_r217201465 and force-pushed.

**Comment by promag** [comment]
utACK e4ef4b4.

**Pull request merged**